File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        curr = dummy
        while(n > 0):
            head = head.next
            n -= 1
        
        while(head):
            head = head.next
            curr = curr.next
        
        curr.next = curr.next.next
        return dummy.next

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-2.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {boolean}
     */
    hasCycle(head) {
        let first = head
        let second = head.next
        while(second && second.next){
            first = first.next
            second = second.next.next
            if(first == second){
                return true
            }
        }
        return false
    }
}
